docs/design/2022-10-27-geospatial.md

```diff
@@ -0,0 +1,294 @@
+|   2165 |
+|   2043 |
+|   2041 |
++--------+
+5 rows in set (0.00 sec)
+
+mysql-8.0.31> CREATE TABLE tbl1(id int primary key, g geometry srid 0);
+Query OK, 0 rows affected (0.05 sec)
+
+mysql-8.0.31> CREATE TABLE tbl2(id int primary key, g geometry srid 1);
+ERROR 3548 (SR001): There's no spatial reference system with SRID 1.
+mysql-8.0.31> CREATE TABLE tbl3(id int primary key, g geometry srid 4143);
+Query OK, 0 rows affected (0.06 sec)
+```
+
+#### Phase 1
+
+Implement `ST_SPATIAL_REFERENCE_SYSTEMS`, keeping this compatible with MySQL 8.0.
+
+#### Phase 2
+
+Implement `ST_GEOMETRY_COLUMNS`, keeping this compatible with MySQL 8.0.
+
+#### Phase 3
+
+Implement `ST_UNITS_OF_MEASURE`, keeping this compatible with MySQL 8.0.
+
+### Compatibility
+
+As the parser is updated external tools like Dumpling need to be built with the new parser to be able to validate the schema.
+
+The `sync_diff_inspector` tool might need to be updated to compare and/or ignore geospatial columns.
+
+Downgrading to a release without geospatial datatypes will require dropping all geospatial columns from all tables.
+
+Support for TiFlash is out of scope for now and needs more research.
+
+Geospatial support is not expected to impact security. The `ST_GEOMETRY_COLUMNS` table, when implemented, should limit the visible tables based on the privileges of the user, just like `information_schema.TABLES`.
+
+## Test Design
+
+### Functional Tests
+
+MySQL has a [GIS testsuite](https://github.com/mysql/mysql-server/tree/8.0/mysql-test/suite/gis). This can be used with [mysql-tester](https://github.com/pingcap/mysql-tester) to test functionality and compatibility with MySQL.
+
+When testing the functions special care should be taken to correctly test if it is correctly using the bounding box or the actual shape.
+
+### Compatibility Tests
+
+Compatibility with Dumpling, Lightning, TiCDC and sync-diff-inspector has to be explicitly tested.
+
+Downgrading without geospatial columns should work and should be tested.
+
+## Impacts & Risks
+
+This adds a new dependency: [github.com/twpayne/go-geom](https://pkg.go.dev/github.com/twpayne/go-geom). The BSD-2-Clause license should be fine. The basic geometric types in this package are ok, but the features that do things like comparing features, calculating distances etc. are not (yet) on the same level as [Boost.Geometry](https://github.com/boostorg/geometry). This means that we may have to add more dependencies and/or implement some missing features ourselves.
+
+## Client support
+
+Working with geospatial data is possible with MySQL Client or mycli. However a GUI tool makes many things much easier. MySQL Workbench has [a spatial viewer](https://dev.mysql.com/blog-archive/mysql-workbench-6-2-spatial-data/) since version 6.2. This works with minimal geospatial support in TiDB. However [DBeaver](https://github.com/dbeaver/dbeaver) has a much more advanced [geospatial viewer](https://dbeaver.com/docs/wiki/Working-with-Spatial-GIS-data/).
```

Comment:
Do you mean that DBeaver does not work with the minimal geospatial support in TiDB?